### DISCUSSION TITLE
How to combine 'compose' and 'view' based UI

### DISCUSSION THREAD

**Discussion by yoouyoou** [euphony-io/euphony#163] [Q&A]
Hello.
I tried to implement a view using Compose.
But this is difficult, so I'm going to mix it with 'xml based layout'.

https://developer.android.com/jetpack/compose/interop/interop-apis?hl=ko
I referred to this development document, but I don't know.

I added activity dependency to the gradle file and created the xml file, but I don't know how to display it.

**Comment by zion830** [comment]
If you want to add `compose ui` to `xml view`, just use `ComposeView`
```
<androidx.compose.ui.platform.ComposeView
    android:id="@+id/composeView"
    android:layout_width="wrap_content"
    android:layout_height="wrap_content" />
```
```
findViewById<ComposeView>(R.id.composeView).setContent {
    // Compose ui code
}
```
If you want the opposite,
```
@Composable
fun ComposeViewWithXml() {
    AndroidView(
        modifier = Modifier.fillMaxSize(),
        factory = { context ->
            val view = LayoutInflater.from(context).inflate(R.layout.layout_id, null, false)
            // init view
            view // return the view
        },
        update = { view ->
            // update the view
        }
    )
}
```
Maybe we need to see some code to find an exact reason.
